src/posix/platform/infra_if.cpp

```diff
@@ -726,15 +771,44 @@ otError InfraNetif::DiscoverNat64Prefix(uint32_t aInfraIfIndex)
 }
 #endif // OPENTHREAD_POSIX_CONFIG_NAT64_AIL_PREFIX_ENABLE
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+void InfraNetif::SetInfraNetifForBorderRouting(int aIcmp6Socket)
```

Comment:
is this needed for both BORDER_ROUTING and BACKBONE_ROUTER ? If so should it be under `#if OPENTHREAD_POSIX_CONFIG_INFRA_IF_ENABLE` instead?

Reply:
No. `SetInfraNetifForBorderRouting` is only needed for border routing when dealing with ICMPv6 ND messages. For backbone router, although we usually need OTBR have an on-link address on infra link, it won't automatically be fulfilled unless border routing is enabled or there's another host sending RA PIO on infra link.

Reply:
I see, so this is just setting the icmp6 socket only required by border routing, may update the function name to `setInfraNetifSocketForBorderRouting` ?

Reply:
Sounds good. The code is updated.